### PR TITLE
Add "podman kube generate" command

### PR DESCRIPTION
**Pull request by nicrowe00**
"podman kube generate" creates Kubernetes YAML from Podman containers,
pods or volumes. Users will still be able to use "podman generate
kube" as an alias of "kube generate".

Signed-off-by: Niall Crowe <nicrowe@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
Adds new command, "podman kube generate", which generates Kubernetes YAML
from containers, pods or volumes. The command "podman generate kube" can still be
used and is now an alias of "kube generate"
```

### CODE REVIEW

**Comment by nicrowe00** [comment]
@vrothberg PTAL

**Comment by nicrowe00** [comment]
@containers/podman-maintainers PTAL

File: docs/source/markdown/podman-kube-generate.1.md [hunk 1]
```diff
@@ -213,7 +213,4 @@ status:
 ```
 
 ## SEE ALSO
-**[podman(1)](podman.1.md)**, **[podman-container(1)](podman-container.1.md)**, **[podman-pod(1)](podman-pod.1.md)**, **[podman-kube-play(1)](podman-kube-play.1.md)**
-
-## HISTORY
-December 2018, Originally compiled by Brent Baude (bbaude at redhat dot com)
```
Comment (vrothberg) [inline docs/source/markdown/podman-kube-generate.1.md:219]: I think we should preserve the history.

File: pkg/api/server/register_play.go [hunk 2]
```diff
@@ -78,5 +78,38 @@ func (s *APIServer) registerPlayHandlers(r *mux.Router) error {
 	//     $ref: "#/responses/internalError"
 	r.HandleFunc(VersionedPath("/libpod/play/kube"), s.APIHandler(libpod.PlayKubeDown)).Methods(http.MethodDelete)
 	r.HandleFunc(VersionedPath("/libpod/kube/play"), s.APIHandler(libpod.KubePlayDown)).Methods(http.MethodDelete)
+	// swagger:operation GET /libpod/generate/kube libpod GenerateKubeLibpod
```
Comment (vrothberg) [inline pkg/api/server/register_play.go:81]: Probably worth renaming the file to `register_kube.go`.

File: pkg/bindings/kube/kube.go [hunk 3]
```diff
@@ -94,3 +95,7 @@ func DownWithBody(ctx context.Context, body io.Reader) (*entities.KubePlayReport
 	}
 	return &report, nil
 }
+
+func Generate(ctx context.Context, nameOrIDs []string, options generate.KubeOptions) (*entities.GenerateKubeReport, error) {
```
Comment (vrothberg) [inline pkg/bindings/kube/kube.go:99]: Please add a comment to the function.

File: test/e2e/generate_kube_test.go [hunk 4]
```diff
@@ -58,7 +58,7 @@ var _ = Describe("Podman generate kube", func() {
 		session.WaitWithDefaultTimeout()
 		Expect(session).Should(Exit(0))
 
-		kube := podmanTest.Podman([]string{"generate", "kube", "top"})
+		kube := podmanTest.Podman([]string{"kube", "generate", "top"})
```
Comment (vrothberg) [inline test/e2e/generate_kube_test.go:61]: Are there still instances of `generate kube` left?

Comment (nicrowe00) [reply test/e2e/generate_kube_test.go:61]: Yes there is still instances of generate kube left

**Comment by TomSweeneyRedHat** [comment]
LGTM
but I did  a very high level skim review.

**Comment by rhatdan** [comment]
If I do podman generate kube, does it still work?

**Comment by nicrowe00** [comment]
@rhatdan Yes, podman generate kube should still work.

**Pull request merged**